### ISSUE TITLE
Unresponsive UI and high CPU usage due to "[ts] Unreachable code detected. [7027]"

### ISSUE THREAD

**Issue by pqvst** [Microsoft/vscode#67846]
- VSCode Version: 1.30.2 (1.30.2)
- OS Version: macOS 10.14.2 (18C54)

Steps to Reproduce:
1. Create/open a `.js` source file (standalone or in project)
2. Add a `return` statement at the top of a function
3. Code after the return statement will be flagged as "Unreachable code detected"
4. UI performance degrades significantly and causes high CPU when scrolling or typing when the unreachable code is visible in the editor.

Does this issue occur when all extensions are disabled?: Yes

A significant CPU usage spike is observed when editing the code file (reaching ~50% at times):
![screenshot 2019-02-04 12 45 52](https://user-images.githubusercontent.com/4990657/52206557-df5c4c80-287a-11e9-81d8-427e70356e28.png)

Sometimes during debugging code you may want to short-circuit a function for whatever reason by adding an early return statement. This causes the editor performance to degrade significantly which makes the editor experience really bad. A workaround is to instead comment out the unreachable code, but this is not always desirable. I expect to editor performance to remain good regardless of whether or not there is unreachable code visible on the screen.

### Demo file (test.js):
Create a new file called `test.js` and insert the below code snippet:
```

function foo() {
  return;
  
  while (true) {
    let foo;
    let bar;
    
    if (!foo) {
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
      foo = 'bar';
    }
  }
}


bar() {
  let bar;
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
  bar = 'bar';
}
```

**Comment by dbaeumer** [comment]
@alexandrudima Not sure if this is editor or TS extensions.

**Comment by mjbvz** [comment]
Likely fixed by electron 3.

@pqvst Does this reproduce in the [latest VS Code insiders build](https://code.visualstudio.com/insiders/) with all extensions disabled?

**Comment by mjbvz** [comment]
ok, this fix will be in the VS 1.31 stable build coming out soon
